@here/harp-map-controls/lib/MapControls.ts

```diff
@@ -777,41 +789,73 @@ export class MapControls extends THREE.EventDispatcher {
+            } else if (this.mapView.projection.type === geoUtils.ProjectionType.Spherical) {
+                this.m_rotateGlobeQuaternion
+                    .setFromAxisAngle(
+                        this.m_lastRotateGlobeAxis,
+                        this.m_currentInertialPanningSpeed
+                    )
+                    .normalize();
             }
-            this.m_panDistanceFrameDelta
-                .copy(this.m_lastPanVector)
-                .setLength(this.m_currentInertialPanningSpeed);
         } else {
-            this.m_lastPanVector.copy(this.m_panDistanceFrameDelta);
-            const panDistance = this.m_lastPanVector.length();
-            this.m_currentPanDistanceIndex =
-                (this.m_currentPanDistanceIndex + 1) % USER_INPUTS_TO_CONSIDER;
-            this.m_recentPanDistances[this.m_currentPanDistanceIndex] = panDistance;
-            this.m_lastAveragedPanDistance =
-                this.m_recentPanDistances.reduce((a, b) => a + b) / USER_INPUTS_TO_CONSIDER;
+            let panDistanceOrAngle: number = 0;
+            if (this.mapView.projection.type === geoUtils.ProjectionType.Planar) {
+                panDistanceOrAngle = this.m_lastPanVector
+                    .copy(this.m_panDistanceFrameDelta)
+                    .length();
+            } else if (this.mapView.projection.type === geoUtils.ProjectionType.Spherical) {
+                panDistanceOrAngle = this.m_lastRotateGlobeAngle;
+                this.m_rotateGlobeQuaternion
+                    .setFromAxisAngle(this.m_lastRotateGlobeAxis, this.m_lastRotateGlobeAngle)
+                    .normalize();
+            }
+            this.m_currentPanDistanceOrAngleIndex =
+                (this.m_currentPanDistanceOrAngleIndex + 1) % USER_INPUTS_TO_CONSIDER;
+            this.m_recentPanDistancesOrAngles[
+                this.m_currentPanDistanceOrAngleIndex
+            ] = panDistanceOrAngle;
+            this.m_lastAveragedPanDistanceOrAngle =
+                this.m_recentPanDistancesOrAngles.reduce((a, b) => a + b) / USER_INPUTS_TO_CONSIDER;
         }
 
-        MapViewUtils.pan(
-            this.mapView,
-            this.m_panDistanceFrameDelta.x,
-            this.m_panDistanceFrameDelta.y
-        );
+        if (this.mapView.projection.type === geoUtils.ProjectionType.Planar) {
+            MapViewUtils.panCameraAboveFlatMap(
+                this.mapView,
+                this.m_panDistanceFrameDelta.x,
+                this.m_panDistanceFrameDelta.y
+            );
+        } else if (this.mapView.projection.type === geoUtils.ProjectionType.Spherical) {
+            MapViewUtils.rotateCameraAroundGlobe(
+                this.mapView,
+                this.m_lastRotateGlobeFromVector,
+                this.m_tmpVector3
+                    .copy(this.m_lastRotateGlobeFromVector)
+                    .applyQuaternion(this.m_rotateGlobeQuaternion)
+            );
+            this.mapView.camera.updateMatrixWorld(true);
```

Comment:
If its really needed (no update from outside), I guess it should be encapsulated in MapViewUtils.rotateCameraAroundGlobe() method implementation, it already has mapView reference to access.